types/msg_test.go

```diff
@@ -47,9 +47,7 @@ func TestWasmMsgInstantiateSerialization(t *testing.T) {
 	require.Equal(t, "king", msg.Instantiate.Admin)
 	require.Equal(t, uint64(7897), msg.Instantiate.CodeID)
 	require.Equal(t, []byte(`{"claim":{}}`), msg.Instantiate.Msg)
-	require.Equal(t, Coins{
-		{"stones", "321"},
-	}, msg.Instantiate.Funds)
+	require.Equal(t, Array[Coin]{}, msg.Instantiate.Funds)
```

Comment:
It somewhat confuses me that this used to pass with the `Coins`. The json contains an empty `funds` field, so an empty value seems more reasonable to me here. I guess the old version was producing a `nil`, which made the json marshalling keep the existing value? (pure speculation on my part)

Reply:
I guess it's because the old Coins unmarshaller returned early with a nil error, but also did not change `*c`. So the bug was that `*c = d` is not executed. Seems like we have this bug fixed now. Can we test it for Array (unmarshall elements followed by nil into the same pointer)?

Reply:
Ah yes, that makes much more sense. Yes, will add a test for that.